app/transformations/MemberPaymentsTransformer.scala

```diff
@@ -156,35 +157,46 @@ class MemberPaymentsTransformer @Inject()(
         }
       }
 
+      val normalise = (memberDetails: MemberDetails) => {
+        memberDetails.copy(
+          memberLumpSumReceived = if (memberDetails.memberLumpSumReceived.exists(_.zero)) {
+            None
+          } else {
+            memberDetails.memberLumpSumReceived
+          },
+          pensionAmountReceived =
+            if (memberDetails.pensionAmountReceived.contains(0.0)) None
+            else memberDetails.pensionAmountReceived,
+          totalContributions =
+            if (memberDetails.totalContributions.contains(0.0)) None
+            else memberDetails.totalContributions
+        )
+      }
+
       // Omit memberPSRVersion if member has changed
       // Check for empty member payment sections before comparing members
       // (this is because we send empty records in certain sections for members)
       //
       // Note: This will be changing soon when we re-design statuses
       val memberDetailsWithCorrectVersion: List[MemberDetails] = memberDetailsWithCorrectState.map {
         case (index, currentMemberDetail) =>
-          val optInitialMemberDetail = initialMemberDetails.get(index)
-          val refinedCurrentMember =
-            currentMemberDetail
-              .copy(
-                memberLumpSumReceived =
-                  if (currentMemberDetail.memberLumpSumReceived.exists(_.zero)) None
-                  else currentMemberDetail.memberLumpSumReceived,
-                pensionAmountReceived =
-                  if (currentMemberDetail.pensionAmountReceived.contains(0.0)) None
-                  else currentMemberDetail.pensionAmountReceived,
-                totalContributions =
-                  if (currentMemberDetail.totalContributions.contains(0.0)) None
-                  else currentMemberDetail.totalContributions
+          initialMemberDetails.get(index) match {
+            case None =>
+              currentMemberDetail.copy(memberPSRVersion = None)
+            case Some(initialMemberDetail) =>
+              val normalisedInitialMember = normalise(initialMemberDetail)
+              val normalisedCurrentMember = normalise(currentMemberDetail)
+              val same = normalisedInitialMember == normalisedCurrentMember
+              if (!same) {
+                logger.info(s"member $index has changed, removing memberPSRVersion")
+                if(logger.isDebugEnabled) {
+                  logger.debug(Diff(normalisedInitialMember, normalisedCurrentMember).mkString(" - "))
```

Comment:
prints human readable, field specific diffs - great for debugging issues in the future around recordVersion / memberPsrVersion